hudi-utilities/src/main/java/org/apache/hudi/utilities/sources/JDBCSource.java

```diff
@@ -0,0 +1,238 @@
+        }
+      } else if (!StringUtils.isNullOrEmpty(properties.getString(Config.PASSWORD))) {
+        dataFrameReader = dataFrameReader.option(Config.PASSWORD_PROP, properties.getString(Config.PASSWORD));
+      } else {
+        throw new IllegalArgumentException(String.format("%s cannot be null or empty. ", Config.PASSWORD));
+      }
+//      if (properties.containsKey(Config.EXTRA_OPTIONS)) {
+//        if (!StringUtils.isNullOrEmpty(properties.getString(Config.EXTRA_OPTIONS))) {
+//          LOG.info("Setting {}", Config.EXTRA_OPTIONS);
+//          String[] options = properties.getString(Config.EXTRA_OPTIONS).split(",");
+//          for (String option : options) {
+//            if (!StringUtils.isNullOrEmpty(option)) {
+//              String[] kv = option.split("=");
+//              if (kv.length == 2) {
+//                dataFrameReader = dataFrameReader.option(kv[0], kv[1]);
+//                LOG.info("{} = {} has been set for JDBC pull ", kv[0], kv[1]);
+//              } else {
+//                LOG.warn("Option {} not set because it does not have a value", kv[0], new IllegalArgumentException(
+//                    String.format(" %s should have a corresponding value separated by \"=\"", kv[0])));
+//              }
+//            }
+//          }
+//        }
+//      }
+
+      properties.entrySet().stream().forEach(i-> System.out.println(properties.get(i)));
+      if (properties.getBoolean(Config.IS_INCREMENTAL)) {
+        DataSourceUtils.checkRequiredProperties(properties, Arrays.asList(Config.INCREMENTAL_COLUMN));
+      }
+      return dataFrameReader;
+    } catch (Exception e) {
+      throw new HoodieException(e);
+    } finally {
+      IOUtils.closeStream(passwordFileStream);
+    }
+  }
+
+  @Override
+  protected Pair<Option<Dataset<Row>>, String> fetchNextBatch(Option<String> lastCkptStr, long sourceLimit) {
+   // final String ppdQuery = "(select * from %s where %s >= TIMESTAMP(\"%s\")) rdbms_table";
+     final String ppdQuery = "(select * from %s where %s >= \"%s\") rdbms_table";
+    try {
+      DataSourceUtils.checkRequiredProperties(props, Arrays.asList(Config.URL, Config.DRIVER_CLASS, Config.USER,
+          Config.RDBMS_TABLE_NAME, Config.IS_INCREMENTAL));
+
+      Option<String> lastCheckpoint =
+          lastCkptStr.isPresent() ? lastCkptStr.get().isEmpty() ? Option.empty() : lastCkptStr : Option.empty();
+      boolean isIncremental = props.getBoolean(Config.IS_INCREMENTAL);
+      if (lastCheckpoint.equals(Option.empty())) {
+        LOG.info("No previous checkpoints found.. ");
+        return full(isIncremental);
+      } else {
+        if (StringUtils.isNullOrEmpty(lastCheckpoint.get())) {
+          LOG.warn("Previous checkpoint entry was null or empty. Falling back to full jdbc pull.");
+          return full(isIncremental);
+        } else {
+          return incremental(ppdQuery, lastCheckpoint, isIncremental);
+        }
+      }
```

Comment:
```java
if (lastCheckpoint.isPresent()) {
    if (StringUtils.isNullOrEmpty(lastCheckpoint.get())) {
          LOG.warn("Previous checkpoint entry was null or empty. Falling back to full jdbc pull.");
          return full(isIncremental);
    } else {
		return incremental(ppdQuery, lastCheckpoint, isIncremental);
	}
} else {
	LOG.info("No previous checkpoints found.. ");
    return full(isIncremental);
}

```

maybe would be changed to the code snippet since it is more clear and readable.